File: alumnium/tools/hover_tool.py
from pydantic import BaseModel, Field
from playwright.sync_api import Page
from selenium.webdriver.remote.webdriver import WebDriver

from alumnium.drivers import BaseDriver


class HoverTool(BaseModel):
    """Hover an element."""

    id: int = Field(description="Element identifier (ID)")

    def invoke(self, driver: BaseDriver):
        if isinstance(driver, Page) or isinstance(driver, WebDriver):
            driver.hover(self.id)

File: ts/src/tools/HoverTool.ts
import { BaseDriver } from "../drivers/BaseDriver.js";
import { BaseTool } from "./BaseTool.js";

export class HoverTool extends BaseTool {
  static description = "Hover over an element.";

  id: number;

  constructor(args: { id: number }) {
    super();
    this.id = args.id;
  }

  async invoke(driver: BaseDriver): Promise<void> {
    await driver.hover(this.id);
  }
}
